src/tracer.c

```diff
@@ -162,6 +162,13 @@ static bool check_if_sink(vmi_instance_t vmi, vmi_event_t *event, event_response
 
         if ( s->paddr && s->paddr == rip_pa )
         {
+            if ( s->checker && !s->checker(vmi, s) ) {
```

Comment:
This would then be checking if there is a sink_extra structure defined and issue the callback accordingly. Decide what to do based on the response. In order to process a `CONTINUE` response you don't necessarily need to emulate, for example the callback could just increase RIP to skip the CPUID sink. Only for breakpoint based sinks do you need to have the sink init do anything, like allocate the emulation structure and store the data necessary for emulation.